src/mantine-utils/src/create-polymorphic-component/create-polymorphic-component.ts

```diff
@@ -35,7 +35,11 @@ export function createPolymorphicComponent<
     props: ComponentProps<C>
   ) => React.ReactElement;
 
-  type PolymorphicComponent = _PolymorphicComponent & StaticComponents;
+  type ComponentProperties = Omit<FunctionComponent<ComponentProps<any>>, never>;
```

Comment:
The `Omit<T, never>` here basically removes the function signature of `FunctionComponent`, to only keep the static properties

Reply:
(changed the line to use `React.FunctionComponent` instead of importing FunctionComponent separately)